pkg/scheduler/objects/application.go

```diff
@@ -1037,6 +1108,9 @@ func (sa *Application) AddAllocation(info *Allocation) {
 func (sa *Application) addAllocationInternal(info *Allocation) {
 	// placeholder allocations do not progress the state of the app and are tracked in a separate total
 	if info.placeholder {
+		if resources.IsZero(sa.allocatedPlaceholder) {
+			sa.initPlaceholderTimer()
+		}
```

Comment:
We did discuss this. The point you bring up about sizes etc has nothing to do with placeholders. It happens as part of normal scheduling also. If you ask for things that can never fit anywhere in the cluster it will sit there. That is irrespective of placeholder or real allocations.
To prevent this we would somehow need to predict that the request will never fit, which means we also know exactly what can be auto scaled. Since we do not have that knowledge we cannot do the preemptive fail or reject.

We also only start the timer at this point, after the first allocation, to really time out the placeholder when it starts using resources. If we start it on first scheduling attempt or some other point in time we have an unstable reference point.
The first scheduling cycle might only trigger a scale up or do an allocation on an available node immediately. This would give some apps the full timeout to use the placeholder while other apps only have seconds or a minute left as the scale up was really slow. That behaviour is difficult to explain and troubleshoot. The allocation on the node is a stable point which can be tracked and explained.

I do agree we need this documented in the code also.

Reply:
I added some comments to this part to explain why we are starting the timer at this point